hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeAccumulator.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.fees;
+
+import com.hedera.node.app.hapi.utils.fee.FeeObject;
+import com.hederahashgraph.api.proto.java.HederaFunctionality;
+import com.hederahashgraph.api.proto.java.Query;
+import com.hederahashgraph.api.proto.java.Timestamp;
+
+/**
+ * Interface for fee calculation. Currently, it is only used to compute payments for Queries. It
+ * will be enhanced to be used for transactions as well in the future.
+ */
+public interface FeeAccumulator {
+    FeeObject computePayment(final HederaFunctionality functionality, final Query query, Timestamp now);
```

Comment:
Should the `FeeAccumulator` know what a `Query` is or should the `Query` know how to work with the `FeeAccumulator`?

Reply:
Currently this is just a wrapper for existing fees class, for getting workflow working. We need to discuss if we should change it.

Reply:
https://github.com/hashgraph/hedera-services/issues/5031